scripts/mdbook-forc-documenter/src/plugins.rs

```diff
@@ -0,0 +1,3 @@
+pub fn plugin_commands() -> Vec<String> {
+    vec!["fmt".to_string(), "explore".to_string(), "lsp".to_string()]
```

Comment:
It might be worth adding a note about updating this to the README when new plugins are added?

Alternatively, we could create a `forc-plugins` directory and move these plugins into this directory. Then we could programmatically discover each of these without having to be aware of updating this list.

---

Also, for lists of known constant strings like this, I think something like the following would be more Rusty:

```rust
pub const PLUGIN_COMMANDS: &[&str] = ["fmt", "explore", "lsp"];
```

You might need to specify the `'static` lifetime on the references, but I think the Rust crew were working towards removing the need for specifying the `'static` lifetime in `const`s.

This saves the 4 dynamic allocations required for the 3 `String`s and the `Vec` that contains them. In our case there's no way this would have any noticable impact on our workflow, but might be worth being mindful of.

Reply:
Hmm the problem is when we're calling `forc --help` to get native commands we're getting a variable number of [`possible_forc_commands()`](https://github.com/FuelLabs/sway/pull/1582/files#diff-68d5c1aed8bd2b9759a0a6f05a0b40fda007ae93bb9172c14bbf86c40735c9c5R7) sometimes this includes invalid commands that we call later when we do `forc <COMMAND> --help`, and we discard invalid commands in the script. The reason why I did it like this is so we do not have to manually add/remove commands to a const slice.

Then we're using that as input into `get_contents_from_commands()` which accepts `&Vec<String>` - that's why I chose to return `Vec<String>` for the `plugin_commands()` as well, unless there's a more idiomatic Rust way to do this(?)

Reply:
> Alternatively, we could create a forc-plugins directory and move these plugins into this directory. Then we could programmatically discover each of these without having to be aware of updating this list.

This sounds like a good idea, a new `forc-plugins` in the proj root?

Reply:
> This sounds like a good idea, a new forc-plugins in the proj root?

Yeah that sounds good! Might also have to update the top-level `[workspace]` for the new paths, and perhaps update any references to the old paths in CI.

Reply:
@binggh ah that makes sense if you're using this with another API that is more dynamic!

`get_contents_from_commands` taking a `&Vec` should be addressed (I [left a comment](https://github.com/FuelLabs/sway/pull/1582/files#r876675992) on how and why), however it will still need to take a slice of `String`, and `String` must be dynamically allocated either way so I think it's fine to disregard my suggestion here (or go with the `forc-plugins` directory instead).

Reply:
Ah yeah that comment makes sense, I'll update that + move the plugins into a `forc-plugins` directory as well